Source/WebCore/css/CSSCounterStyleDescriptors.cpp

```diff
@@ -60,109 +69,167 @@ CSSCounterStyleDescriptors::Ranges translateRangeFromStyleProperties(const Style
     return result;
 }
 
-static String symbolToString(const CSSValue* value)
+
+static CSSCounterStyleDescriptors::Symbol symbolFromCSSValue(const CSSValue* value)
 {
     if (!value || !value->isPrimitiveValue())
         return { };
 
     auto& primitiveValue = downcast<CSSPrimitiveValue>(*value);
-    return primitiveValue.stringValue();
+    return { primitiveValue.isCustomIdent(), primitiveValue.stringValue() };
 }
 
-CSSCounterStyleDescriptors::AdditiveSymbols translateAdditiveSymbolsFromStyleProperties(const StyleProperties& properties)
+CSSCounterStyleDescriptors::Symbol symbolFromCSSValue(RefPtr<CSSValue> value)
+{
+    return symbolFromCSSValue(value.get());
+}
+
+static CSSCounterStyleDescriptors::Name nameFromCSSValue(const CSSValue* value)
+{
+    if (!value || !value->isPrimitiveValue())
+        return { };
+
+    auto& primitiveValue = downcast<CSSPrimitiveValue>(*value);
+    return makeAtomString(primitiveValue.stringValue());
+}
+
+CSSCounterStyleDescriptors::Name nameFromCSSValue(RefPtr<CSSValue> value)
+{
+    return nameFromCSSValue(value.get());
+}
+
+static CSSCounterStyleDescriptors::AdditiveSymbols additiveSymbolsFromStyleProperties(const StyleProperties& properties)
 {
     auto value = properties.getPropertyCSSValue(CSSPropertyAdditiveSymbols);
     if (!value)
         return { };
+    return additiveSymbolsFromCSSValue(WTFMove(value));
+}
 
+CSSCounterStyleDescriptors::AdditiveSymbols additiveSymbolsFromCSSValue(RefPtr<CSSValue> value)
+{
+    if (!value)
+        return { };
```

Comment:
It should be sufficient to null test in one place only. If the caller is responsible then the `additiveSymbolsFromCSSValue` could take `Ref<CSSValue>` 

This comment applies to other similar functions in this patch.